### ISSUE TITLE
Max value is 0 when being on All Accounts and clicking send, having default account empty

### ISSUE THREAD

**Issue by anastasiyaig** [status-im/status-app#16784]
### Description

![Image](https://github.com/user-attachments/assets/aef8b2e9-42cd-4929-9b8a-8ae0142a098a)

1. create wallet account with funds
2. open wallet
3. click All accounts tab
4. click asset you have balance for, for example ETH
5. click send button

Result: MAX amount is 0, cant sent

https://github.com/user-attachments/assets/fb56501b-745e-4c18-9ddc-102b0af7e6f7

**Comment by Generalresolution03** [comment]
Hello @anastasiyaig 
can you discuss this issue with the official support?

**Comment by anastasiyaig** [comment]
perhaps we need to disable send button for All accounts ? @benjthayer how to mitigate this?

**Comment by benjthayer** [comment]
> perhaps we need to disable send button for All accounts ? [@benjthayer](https://github.com/benjthayer) how to mitigate this?

Hey @anastasiyaig the current logic is that when the Send dialog is invoked via All accounts -> Token detail, we default the account selection to the first in the users Wallet list. Which is why sometimes the max will be 0 even if the user holds that asset in other accounts - its all down to which account we default to.

We could default in this scenario to the account that holds the most of the relevant asset but it was thought at the time that creating more extensive logic for which account is selected would be more dev work and also cause some confusion to the user if they are not clear on why the default account keeps changing.

cc @Khushboo-dev-cpp

**Comment by anastasiyaig** [comment]
@benjthayer but when you click Send when being on a screen with chart, like ETH, then we dont default to set "from" to any of accounts. pls check on the video

**Comment by benjthayer** [comment]
> [@benjthayer](https://github.com/benjthayer) but when you click Send when being on a screen with chart, like ETH, then we dont default to set "from" to any of accounts. pls check on the video

Account 1 is the defaulted account no? Sorry think I might be misunderstanding the issue! 😂

![Image](https://github.com/user-attachments/assets/89c8160d-8967-41ec-be27-f2699dfca2e7)

**Comment by anastasiyaig** [comment]
no you are right :D you are correct, its fine, we indeed need to think of how we can handle the case when default account is empty :D

**Comment by Khushboo-dev-cpp** [comment]
> e current logic is that when the Send dialog is invoked via All accounts -> Token detail, we default the account selection to the first in the users Wallet list. Which is why sometimes the max will be 0 even if the user holds that asset in other accounts - its all down to which account we default to.
> 
> We could default in this scenario to the account that holds the most of the relevant asset but it was thought at the time that creating more extensive logic for which account is selected would be more dev work and also cause some confusion to the user if they are not clear on why the default account keeps changing.

@benjthayer @anastasiyaig yes we thought, switching accounts based on which accounts holds most tokens might be confusing, hence this logic. @benjthayer maybe we can choose the account to most balance and show a toast message that says account with most balance has been selected as recipient.?

**Comment by anastasiyaig** [comment]
@Khushboo-dev-cpp we need to default to account with balance that you can send **from**, so sender, not recipient

**Comment by benjthayer** [comment]
> > e current logic is that when the Send dialog is invoked via All accounts -> Token detail, we default the account selection to the first in the users Wallet list. Which is why sometimes the max will be 0 even if the user holds that asset in other accounts - its all down to which account we default to.
> > We could default in this scenario to the account that holds the most of the relevant asset but it was thought at the time that creating more extensive logic for which account is selected would be more dev work and also cause some confusion to the user if they are not clear on why the default account keeps changing.
> 
> [@benjthayer](https://github.com/benjthayer) [@anastasiyaig](https://github.com/anastasiyaig) yes we thought, switching accounts based on which accounts holds most tokens might be confusing, hence this logic. [@benjthayer](https://github.com/benjthayer) maybe we can choose the account to most balance and show a toast message that says account with most balance has been selected as sender.?

Yeah Im inclined to agree, @Khushboo-dev-cpp thinking on it. I'm not sure we need the toast though - when the user clicks on the account dropdown, they will see that it is the account with the highest value of the selected token that has been selected for them:
https://www.figma.com/design/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=25193-64032&t=1ThWg0e78q6kxv8z-4

**Comment by anastasiyaig** [comment]
this one is not an issue anymore
